packages/tinacms/src/cache/node-cache.ts

```diff
@@ -1,25 +1,40 @@
 import type { Cache } from './index'
 
+//? Note - These will need to be changed from using require to import when we eventually move to ESM
 // makeCacheDir creates the cache directory if it doesn't exist
 const makeCacheDir = async (dir: string, fs: any) => {
-  const path = await import('node:path')
-  const os = await import('node:os')
+  const path = require('node:path')
+  const os = require('node:os')
 
-  const parts = dir.split(path.sep).filter(Boolean)
+  // Ensure that `dir` is a valid string
+  if (typeof dir !== 'string' || !dir.trim()) {
+    throw new Error('Invalid directory path')
+  }
 
+  const pathParts = dir.split(path.sep)
+  const cacheHash = pathParts[pathParts.length - 1]
   let cacheDir = dir
-  // check if the root directory exists, if not then create create in tmp and return new path
-  if (!fs.existsSync(path.join(path.sep, parts[0]))) {
-    cacheDir = path.join(os.tmpdir(), parts[parts.length - 1])
+
+  // ["Users", "jackpettit", "Developer", "tina", "tina-cloud-starter", "tina", "__generated__", "".cache", "1729659325506"]
+  // "/tmp/17.../"
+
+  if (!fs.existsSync(dir)) {
```

Comment:
This won't work as `dir` won't exist yet. We should check if `.../.cache` exists and if it doesn't, resort to `/tmp`

Reply:
`dir` is a parameter to the function (line 5 after changes), and the existence of a valid value is checked on line 10 (after changes).

Reply:
Sorry @ncn-ssw, I should have been more clear. The system directory that `dir` references won't exist therefor `existsSync` will always result in false